pkg/sql/logictest/testdata/logic_test/cluster_settings

```diff
@@ -293,10 +293,21 @@ skipif config 3node-tenant-default-configs
 statement error cannot use this statement to access cluster settings in system tenant
 ALTER TENANT [1] SET CLUSTER SETTING server.mem_profile.total_dump_size_limit='10M'
 
+skipif config 3node-tenant-default-configs
+statement error cannot set 'version' for all tenants
+ALTER TENANT ALL SET CLUSTER SETTING version='23.1'
+
 skipif config 3node-tenant-default-configs
 statement ok
 SELECT crdb_internal.create_tenant(10)
 
+# While not specifically useful, we continue to allow setting the
+# `version` key for a specific virtual cluster, in case it's necessary
+# to overwrite the tenant's view of the storage cluster's version.
```

Comment:
I know I raised this use case, but I wonder if it is actually ever going to happen.  Perhaps it is safe to disallow it until we have a real use case.

Reply:
Yeah, I had the same thought, especially there's still a workound if for some reason this was necessary (using a direct `INSERT` syntax instead of `ALTER VIRTUAL CLUSTER`). Updated.